File: scalafmt-interfaces/jvm/src/main/java/org/scalafmt/interfaces/ScalafmtSessionFactory.java
package org.scalafmt.interfaces;

import java.nio.file.Path;

public interface ScalafmtSessionFactory {

    /**
     * Create a ScalafmtSession to format a batch of files using fixed configuration.
     * @param config location of the configuration file
     * @return a new session instance
     */
    ScalafmtSession createSession(Path config);

}

File: scalafmt-interfaces/js/src/main/scala/org/scalafmt/interfaces/ScalafmtSessionFactory.scala
package org.scalafmt.interfaces

import java.nio.file.Path

trait ScalafmtSessionFactory {

  /** Create a ScalafmtSession to format a batch of files using fixed
    * configuration.
    * @param config
    *   location of the configuration file
    * @return
    *   a new session instance
    */
  def createSession(config: Path): ScalafmtSession
}
